v7/wordpress_compiler/wordpress/wordpress.py

```diff
@@ -0,0 +1,210 @@
+        # (not even sure if it's really implementable...)
+        raise NotImplementedError()
+
+    def add_filter(self, tag, filter_function, priority=10):
+        if tag not in self.__filters:
+            self.__filters[tag] = list()
+        f = self.__filters[tag]
+        # find where to insert priority
+        i = 0
+        while i < len(f) and f[i][0] < priority:
+            i += 1
+        if i < len(f) and f[i][0] > priority:
+            f.insert(i, (priority, list()))
+        elif i == len(f):
+            f.append((priority, list()))
+        f[i][1].append(filter_function)
+
+    def filter(self, tag, data, context):
+        if tag not in self.__filters:
+            return data
+        for prio, fs in self.__filters[tag]:
+            for f in fs:
+                data = f(data, context)
+        return data
+
+    def register_shortcode(self, tag, function):
+        self.__shortcodes.register_shortcode(tag, function)
+
+    def unregister_shortcode(self, tag):
+        self.__shortcodes.unregister_shortcode(tag)
+
+    def do_shortcode(self, data):
+        return self.__shortcodes.do_shortcode(data)
+
+    def set_site(self, site):
+        super().set_site(site)
+        self._register_plugins()
+
+    def __formatData(self, data, context, source=None):
+        output = self.filter("the_content", data, context)
+        left_shortcodes = self.__shortcodes.get_containing_shortcodes_set(output)
+        if len(left_shortcodes) > 0 and source is not None:
+            _LOGGER.warning("The post '" + source + "' still contains shortcodes: " + str(left_shortcodes))
+        return output
+
+    def compile_to_string(self, source_data):
+        context = Context(hash(source_data))
+        return self.__formatData(source_data, context)
+
+    def _read_extra_deps(self, post):
+        dep_path = post.base_path + '.dep'
+        if os.path.isfile(dep_path):
+            with io.open(dep_path, 'rb') as file:
+                result = pickle.load(file)
+                if type(result) == tuple and len(result) == 4:
+                    return result
+        return ([], [], [], [])
+
+    def register_extra_dependencies(self, post):
+        post.add_dependency(lambda: self._read_extra_deps(post)[0], 'fragment')
+        post.add_dependency(lambda: self._read_extra_deps(post)[1], 'page')
+        post.add_dependency_uptodate(lambda: self._read_extra_deps(post)[2], True, 'fragment')
+        post.add_dependency_uptodate(lambda: self._read_extra_deps(post)[3], True, 'page')
+
+    def _write_deps(self, context, dest):
+        deps_path = dest + '.dep'
+        if context.has_dependencies():
+            data = (context.get_file_dependencies_fragment(), context.get_file_dependencies_page(),
+                    context.get_uptodate_dependencies_fragment(), context.get_uptodate_dependencies_page())
+            with io.open(deps_path, "wb") as file:
+                pickle.dump(data, file, pickle.HIGHEST_PROTOCOL)
```

Comment:
Please use JSON or some sane serialization format. Pickle is arbitrary code execution.